### ISSUE TITLE
Create Game Screen

### ISSUE THREAD

**Issue by 03hchen** [GTBitsOfGood/jennifer-anns-group#29]
## Description
The admin should be able to create a game with custom name, game build, video trailer link, description, parenting guide, lesson plan, answer key, themes, accessibility tags, and custom tags. Create a page at `/admin/create-game` that follows the [Figma design here](https://www.figma.com/file/9qcNl3ZnyTQqnIcXMxiFOw/Jennifer-Annes-S24?type=design&node-id=11094-4988&mode=design&t=PjWZpwzLImNflDWX-4). 

Only the admin should have access to this page. Other users who attempt to access this page should be redirected to the home page `/`. Look into the existing code at `src/components/HOC/PageAccess.tsx` for how to do this.

For now, we do not have to implement a Preview option. Create a "Publish" instead of "Preview" button (as the design shows) that adds the game to the database and redirects the user to the game gallery at `/games` (this page doesn't exist rn but will soon).

ALL existing themes, accessibility tags, and tags can be found in the themes and tags collections. **All** options should appear on the Create a Game page, and the user should have the option to select which one the game will have. The user does not have the option to create a new theme or tag on this page.

## Acceptance Criteria
- [ ] The create a game page is created following the requirements above
- [ ] Clicking "Publish" creates a game in the database and redirects the user
- [ ] The page follows the Figma design
- [ ] Error fields appear for fields that are required if left blank

## Other Notes
* You do **NOT** have to implement PDF upload or game build functionality or design for now (screenshots below). If you have time you can create the frontend designs for them (optional though), but they definitely don't have to save anything to the database yet. 
<img width="621" alt="Screenshot 2024-02-20 at 10 25 40 AM" src="https://github.com/GTBitsOfGood/jennifer-anns-group/assets/72580357/89bac554-af37-440c-9c7e-0237732b47cb">
<img width="345" alt="Screenshot 2024-02-20 at 10 25 49 AM" src="https://github.com/GTBitsOfGood/jennifer-anns-group/assets/72580357/1e28ec8a-b8c7-4995-a1dd-81d0ac77c44d">

* A lot of the elements on the page are similar to signup/login, profile modal, and specific game screen which are all already implemented, so make sure to take a look at those pages and reuse components!
* Feel free to use any component library that may help such as [Chakra](https://chakra-ui.com/docs/components/modal) or [shadcn](https://ui.shadcn.com/)
* Make sure to use Tailwind for styling!
* There's a lot of nuances with adding game builds for this page so let us know if you need clarifications about the design and don't be afraid to reach out for help!
